### PR TITLE
Configure Snowplow analytics based on cookie consent (Fixes #12693)

### LINKED ISSUE #12693: [pocket] Snowplow analytics config should differ based on cookie prefs
From Anthony Liddle on the Pocket team:

We wait until OneTrust has loaded before initializing Snowplow to know which config to use.

Here’s the config we use to initialize snowplow if the user allows cookies:

```
{
  appId: <SNOWPLOW_APP_ID>,
  platform: 'web',
  eventMethod: 'beacon',
  respectDoNotTrack: false,
  stateStorageStrategy: 'cookieAndLocalStorage',
  contexts: { webPage: true, performanceTiming: true },
  postPath: <SNOWPLOW_POST_PATH>
}
```

And here’s the config we use when they’re opted out of cookies:

```
{
  appId: <SNOWPLOW_APP>,
  platform: 'web',
  eventMethod: 'post',
  respectDoNotTrack: false,
  stateStorageStrategy: 'none',
  contexts: { webPage: true, performanceTiming: true },
  anonymousTracking: { withServerAnonymisation: true },        
  postPath: <SNOWPLOW_POST_PATH>
}
```

If a user turns off cookie tracking during a session, we clear all cookies user data using the built in Snowplow method `clearUserData`, and then reload the page.

Note that the `eventMethod` also changes between the two configs from `beacon` to `post`.

### PR DESCRIPTION
**Pull request by alexgibson**
## One-line summary

This PR changes the way we load snowplow analytics for Pocket, based on the consent provided by the OneTrust cookie banner.  Similar to https://github.com/mozilla/bedrock/issues/12670, but Snowplow has some better privacy features compared to GA.

## Significant changes and points to review

- This PR also updates the existing analytics code and adds JS tests for the logic. It's worth testing everything still looks ok.
- I'm also updating the snowplow JS bundle to v2.18, as this is needed when calling `clearUserData`.

## Issue / Bugzilla link

#12693
#12689

## Testing

Demo link: https://www-demo1.tekcopteg.com/en/about/

To test locally, you'll need to set the following `.env` variables:

```
GTM_CONTAINER_ID=G-NFR9Y40GD3
GOOGLE_ANALYTICS_ID=UA-370613-9
```

- [x] On initial load, cookie consent banner is shown and both GA and Snowplow scripts are loaded.
- [x] Click "Manage" -> "Analytics Cookies" -> "Disable" -> "Confirm".
- [x] Page should automatically refresh after opting out.
  - [x] GA should not be loaded after page refresh.
  - [x] Snowplow should still be loaded after refresh but in privacy respecting mode. 
- [x] Opting back in (via Cookie Preferences link in the footer) should load analytics again (requires a manual refresh).

### CODE REVIEW

**Comment by alexgibson** [comment]
Successful test run: https://gitlab.com/mozmeao/www-config/-/pipelines/772301117

File: media/js/pocket/analytics.es6.js [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+            .getElementsByTagName('html')[0]
+            .getAttribute('data-snowplow-connect-url');
+
+        if (!SNOWPLOW_APP_ID || !SNOWPLOW_SCRIPT_SRC || !SNOWPLOW_CONNECT_URL) {
+            return;
+        }
+
+        function getUserData() {
+            const data = {
+                hashed_user_id: Mozilla.Cookies.getItem('a95b4b6'), // set only when logged-in through the Pocket app.
```
Comment (maureenlholland) [inline media/js/pocket/analytics.es6.js:91]: great comments, thanks

File: tests/unit/spec/pocket/analytics.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                eventMethod: 'beacon',
+                stateStorageStrategy: 'cookieAndLocalStorage',
+                anonymousTracking: false
+            });
+            expect(window.PocketAnalytics.loadGA).toHaveBeenCalled();
+            expect(window.PocketAnalytics.loaded).toBeTrue();
+            expect(window.PocketAnalytics.reloadPage).not.toHaveBeenCalled();
+        });
+
+        it('should not GA when cookie consent is not given', function () {
```
Comment (maureenlholland) [inline tests/unit/spec/pocket/analytics.js:42]:
```suggestion
        it('should not load GA when cookie consent is not given', function () {
```

**Review by maureenlholland — APPROVED** [review]
r+ ✅  everything looks good

**Review by maureenlholland — COMMENTED** [review]
@alexgibson js tests failing in CI

**Comment by alexgibson** [comment]
@maureenlholland thanks, updated 👍

**Review by maureenlholland — APPROVED** [review]
r++ 🟢

**Pull request merged**